### PR TITLE
client events will now be unsubscribed during unmount

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

`XrplClient` events were not being cleaned up when the `<Streams>`
component was unmounted.  This was causing a memory leak and prevented
the hash section of a ledger from being rendered upon reentering the
ledgers page. This also caused a similar problem on the validators page.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

### Explorer Page
1. Enter Explorer page
2. Navigate to Token page
3. Navigate to Explorer page
4. Hashes for Tokens will render 

### Network Page
1. Enter Validators page
2. Navigate to Explorer page
3. Navigate to Validators page
4. Validator hexes will render

### CODE REVIEW

**Comment by mvadari** [comment]
Looks like this needs to be rebased.

**Comment by ckniffen** [comment]
I will rebase once #129 is merged.

**Comment by JST5000** [comment]
> I will rebase once #129 is merged. Just wanted to get this out there.

If you don't want to wait to rebase, you can also just point this PR at #129 instead of staging and then rebase onto #129 :)

**Comment by mvadari** [comment]
> I will rebase once #129 is merged. Just wanted to get this out there.

It's hard to review this PR because the changes from #129 are here as well. You can set the base branch for the PR as the branch in #129 though

File: src/containers/shared/components/Streams.jsx [hunk 1]
```diff
@@ -139,7 +138,14 @@ class Streams extends Component {
       streams: ['ledger', 'validations'],
     });
 
-    this.mounted = false;
+    if (this.onledgerWrapper) {
```
Comment (khancode) [inline src/containers/shared/components/Streams.jsx:141]:
```suggestion
    if (this.onLedgerWrapper) {
```
I think the casing should be fully camel case for this and all other `this.on<Something>` vars.

File: src/containers/Network/test/validators.test.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ const WS_URL = 'ws://localhost:1234';
 
 describe('Validators Tab container', () => {
   let server;
-  let client;
+  let client = new MockWsClient(WS_URL);
```
Comment (mvadari) [inline src/containers/Network/test/validators.test.js:27]: It's defined in the `beforeEach`, it doesn't need to be defined up here.

**Review by mvadari — APPROVED** [review]

File: src/containers/shared/components/Streams.jsx (hunk 1 above)
Comment (ckniffen) [reply src/containers/shared/components/Streams.jsx:141]: I was using the convention already set in this file what it wraps is called `onledger`.

File: src/containers/Network/test/validators.test.js (hunk 2 above)
Comment (ckniffen) [reply src/containers/Network/test/validators.test.js]: Removed.  It had been throwing errors for me at one point saying it was undefined but that was before some of your changes.

File: src/containers/shared/components/Streams.jsx (hunk 1 above)
Comment (khancode) [reply src/containers/shared/components/Streams.jsx:141]: Ah I see, ok 👍

**Review by khancode — APPROVED** [review]

**Pull request merged**